### ISSUE TITLE
TraceSourceTests.TraceSourceTests.SwitchLevel failure in CI

### ISSUE THREAD

**Issue by stephentoub** [dotnet/corefx#1205]
```
Discovering: System.Diagnostics.TraceSource.Tests
  Discovered:  System.Diagnostics.TraceSource.Tests
  Starting:    System.Diagnostics.TraceSource.Tests
     System.Diagnostics.TraceSourceTests.TraceSourceTests.SwitchLevel(sourceLevel: All, messageLevel: Critical, expected: 1) [FAIL]
        Assert.Equal() Failure
        Expected: 1
        Actual:   0
        Stack Trace:
           d:\j\workspace\dotnet_corefx_windows_release_prtest\src\System.Diagnostics.TraceSource\tests\TraceSourceClassTests.cs(49,0): at System.Diagnostics.TraceSourceTests.TraceSourceTests.SwitchLevel(SourceLevels sourceLevel, TraceEventType messageLevel, Int32 expected)
  Finished:    System.Diagnostics.TraceSource.Tests
```

**Comment by stephentoub** [comment]
FYI, @MarkPflug

**Comment by MarkPflug** [comment]
What is the "prtest" environment? I've looked through the wiki and I can't find any info about the CI build environment.

**Comment by akoeplinger** [comment]
@MarkPflug It is part of the CI that tests pull requests: http://dotnet-ci.cloudapp.net/view/All/

**Comment by stephentoub** [comment]
I took a quick look at this, and I believe the cause is a race between the tests in TraceClassTests and the SwitchLevel test in TraceSourceTests.  The former makes calls to Trace.Refresh(), which iterates through all registered TraceSource and Switch instances and resets them to their initial state, including resetting the switch level.  As such, if Refresh is called during the SwitchLevel test, it could reset the switch level to one where the traced event won't be recorded, causing the failed assertion.  I wrote a quick test and was able to repro this.  Since #1210 disables test case parallelism, that should fix this issue.
